packages/testing/src/createGatewayTester.ts

```diff
@@ -0,0 +1,143 @@
+  name: string;
+  /** The subgraph schema. */
+  schema: GraphQLSchema | IExecutableSchemaDefinition;
+  /** An optional GraphQL Yoga server instance that runs the {@link schema built subgraph}. */
+  yoga?: (schema: GraphQLSchema) => YogaServerInstance<any, any>;
+}
+
+export type GatewayTesterConfig<
+  TContext extends Record<string, any> = Record<string, any>,
+> = GatewayConfigSchemaBase<TContext> &
+  (
+    | {
+        // gateway
+        supergraph: UnifiedGraphConfig;
+      }
+    | {
+        // gateway (composes subgraphs)
+        subgraphs: GatewayTesterSubgraphConfig[];
+      }
+  );
+// TODO: proxy mode
+// TODO: subgraph mode
+
+export interface GatewayTester<
+  TContext extends Record<string, any> = Record<string, any>,
+> extends AsyncDisposable {
+  runtime: GatewayRuntime<TContext>;
+  fetch: typeof fetch;
+  execute(args: {
+    query: string;
+    variables?: Record<string, unknown>;
+    operationName?: string;
+    extensions?: Record<string, unknown>;
+    headers?: Record<string, string>;
+  }): Promise<MaybeAsyncIterable<ExecutionResult<any>>>;
+}
+
+export function createGatewayTester<
+  TContext extends Record<string, any> = Record<string, any>,
+>(config: GatewayTesterConfig<TContext>): GatewayTester<TContext> {
+  //
+  let runtime: GatewayRuntime<TContext>;
+  if ('supergraph' in config) {
+    runtime = createGatewayRuntime(config);
+  } else {
+    // compose subgraphs and create runtime
+    const subgraphs = config.subgraphs.reduce(
+      (acc, subgraph) => {
+        const url = `http://${subgraph.name}/graphql`;
+        const schema =
+          'typeDefs' in subgraph.schema
+            ? makeExecutableSchema(subgraph.schema)
+            : subgraph.schema;
+        return {
+          ...acc,
+          [url]: {
+            name: subgraph.name,
+            url,
+            schema,
+            yoga:
+              subgraph.yoga?.(schema) ||
+              createYoga({ schema, maskedErrors: false, logging: false }),
+          },
+        };
+      },
+      {} as Record<
+        string,
+        {
+          name: string;
+          url: string;
+          schema: GraphQLSchema;
+          yoga: YogaServerInstance<any, any>;
+        }
+      >,
+    );
+    runtime = createGatewayRuntime({
+      ...config,
+      supergraph: getUnifiedGraphGracefully(Object.values(subgraphs)),
+      plugins: (ctx) => [
+        useCustomFetch((url, options, context, info) => {
+          const subgraph = subgraphs[url];
+          if (!subgraph) {
+            throw new Error(`Subgraph for URL "${url}" not found`);
+          }
+          return subgraph.yoga!.fetch(
+            // @ts-expect-error TODO: url can be a string, not only an instance of URL
+            url,
+            options,
+            context,
+            info,
+          );
+        }),
+        ...(config.plugins?.(ctx) || []),
+      ],
+    });
+  }
+
+  const runtimeExecute = buildHTTPExecutor({
+    fetch: runtime.fetch,
```

Comment:
Let's add `endpoint` here to make Bun happy.
https://github.com/graphql-hive/gateway/actions/runs/18780606631/job/53585575434?pr=1625#step:4:6249
```suggestion
    endpoint: 'http://localhost/graphql',
    fetch: runtime.fetch,
```